parser.js

```diff
@@ -1,91 +1,114 @@
-const urlparse = require('url');
-const {dom, rule, ruleset} = require('fathom-web');
+const {makeUrlAbsolute, parseUrl} = require('./url-utils');
 
-function makeUrlAbsolute(base, relative) {
-  const relativeParsed = urlparse.parse(relative);
-
-  if (relativeParsed.host === null) {
-    return urlparse.resolve(base, relative);
-  }
-
-  return relative;
-}
-
-function getProvider(url) {
-  return urlparse.parse(url)
-    .hostname
+function getProvider(host) {
+  return host
     .replace(/www[a-zA-Z0-9]*\./, '')
     .replace('.co.', '.')
     .split('.')
     .slice(0, -1)
     .join(' ');
 }
 
-function buildRuleset(name, rules, processors) {
-  const reversedRules = Array.from(rules).reverse();
-  const builtRuleset = ruleset(...reversedRules.map(([query, handler], order) => rule(
-    dom(query),
-    node => [{
-      score: order,
-      flavor: name,
-      notes: handler(node),
-    }]
-  )));
-
+function buildRuleSet(ruleSet) {
   return (doc, context) => {
-    const kb = builtRuleset.score(doc);
-    const maxNode = kb.max(name);
+    let maxScore = 0;
+    let maxValue;
+
+    for (let currRule = 0; currRule < ruleSet.rules.length; currRule++) {
+      const [query, handler] = ruleSet.rules[currRule];
+
+      const elements = Array.from(doc.querySelectorAll(query));
 
-    if (maxNode) {
-      let value = maxNode.flavors.get(name);
+      if(elements.length) {
+        for (const element of elements) {
+          let score = ruleSet.rules.length - currRule;
```

Comment:
Actually, in relation to my earlier comment about re-evaling `.length` in each loop iteration, you could just declare a `const ruleLength = ruleSet.rules.length` before all the loops and just refer to that.